core/arch/arch.h

```diff
@@ -215,7 +215,7 @@ reg_spill_tls_offs(reg_id_t reg);
 #define OPSZ_SAVED_XMM (YMM_ENABLED() ? OPSZ_32 : OPSZ_16)
 #define OPSZ_SAVED_ZMM OPSZ_64
 #define REG_SAVED_XMM0 (YMM_ENABLED() ? REG_YMM0 : REG_XMM0)
-#define OPSZ_SAVED_OPMASK OPSZ_8
+#define OPSZ_SAVED_OPMASK (proc_has_feature(FEATURE_AVX512BW) ? OPSZ_8 : OPSZ_2)
```

Comment:
Is there a way to have an automated test for different processor features: I guess w/o a full emulator all we could test is init time code, but that may still be useful.

Reply:
No extra test is required. We had never run it on a processor with AVX512F but no AVX512BW. Had almost any test run on such a processor, it would have failed, even on Linux.

Reply:
That is my point: we can't rely on just happening to have every processor variation sitting around for ad-hoc testing.  I'm wondering if there is some way of systematically testing processor features without having to actually run it on all these processor types, to find all these bugs like this one and like #3949 during init time: like, run under QEMU w/ a variety of fake processors or sthg.

Reply:
I filed #3953.

Reply:
I see. In this specific case, since AVX512BW is backward compatible, you could add a switch and just fake while still running on a AVX512BW machine.